sycl/source/handler.cpp

```diff
@@ -1333,7 +1333,10 @@ void handler::ext_oneapi_graph(
     }
     // Store the node representing the subgraph in the handler so that we can
     // return it to the user later.
-    MSubgraphNode = ParentGraph->addSubgraphNodes(GraphImpl->getSchedule());
+    // The nodes of the subgraph are duplicated when added to its parents.
+    // This avoids changing properties of the graph added as a subgraph.
+    MSubgraphNode =
+        ParentGraph->duplicateAndAddSubgraphNodes(GraphImpl->getGraphImpl());
```

Comment:
I think the name of this function is a bit confusing when you see it in use. The fact that nodes are duplicated seems like an implementation detail that this code is not really concerned with.

Reply:
I like the idea to keep two methods for now: with and without node duplication.
But I understand that we prefer not to expose this level of detail in the (public) name of this method. I thus made it private.